scripts/buildkite/golint.sh

```diff
@@ -2,20 +2,32 @@
 
 set -ex
 
+verify_no_files_changed () {
+  # intentionally capture stderr, so status-errors are also PR-failing.
+  # in particular this catches "dubious ownership" failures, which otherwise
+  # do not fail this check and the $() hides the exit code.
+  if [ -n "$(git status --porcelain  2>&1)" ]; then
+    echo "There file changes after applying your diff and performing a build."
+    echo "Please run this command and commit the changes:"
+    echo "\tmake tidy && make copyright && make go-generate && make fmt && make lint"
```

Comment:
nit: though it works in zsh in Mac, I'm not sure it always works with bash to skip -e and still have escape-symbols interpolation. I recommend echo -e "\tmake tidy ...."

Reply:
Changed to use `printf` instead, as `echo -e` prints the `-e` in `/bin/sh` (at least on my mac). This is also a suggested change by shellcheck